src/sagemaker/session.py

```diff
@@ -855,6 +864,12 @@ def train(  # noqa: C901
         inferred_resource_config = update_nested_dictionary_with_values_from_config(
             resource_config, TRAINING_JOB_RESOURCE_CONFIG_PATH, sagemaker_session=self
         )
+        environment = resolve_value_from_config(
+            direct_input=environment,
+            config_path=TRAINING_JOB_ENVIRONMENT_PATH,
+            default_value={},
```

Comment:
for this method, 
```
        environment=None,
```

so we should keep the default value None unless we have a reason to change it